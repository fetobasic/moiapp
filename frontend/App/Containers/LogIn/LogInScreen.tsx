import React, { useCallback, useEffect, useRef, useState } from 'react';
import {
  Text,
  KeyboardAvoidingView,
  StyleSheet,
  TextInput as TextInputRN,
  Animated,
  Keyboard,
  Pressable,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { useIsFocused } from '@react-navigation/native';

import { ApplicationStyles, isIOS, Metrics, Fonts, Colors } from 'App/Themes';

import { TextInput, TextInputPassword, ButtonSimple, Button, ValidationErrorList } from 'App/Components';
import AppLogo from 'App/Images/appLogo.svg';
import CloseIcon from 'App/Images/Icons/close.svg';
import { useAppDispatch, useAppSelector, useEvents, useLoginAnimation, useLoginNavigation, useMount } from 'App/Hooks';
import { validate } from 'App/Services/Validator';
import { authActions } from 'App/Store/Auth';
import { LogIn } from 'App/Types/Validation';
import { loginRules } from './validateRules';
import AppConfig from 'App/Config/AppConfig';
import { TextInputRef } from 'App/Components/TextInput';
import { LoginStackParamList } from 'App/Types/NavigationStackParamList';

type Props = NativeStackScreenProps<LoginStackParamList, 'LogIn'>;

function LogInScreen({ route }: Props) {
  const navigation = useLoginNavigation('LogIn');
  const dispatch = useAppDispatch();
  const isFocused = useIsFocused();
  const { track, identify } = useEvents();

  const emailInputRef = useRef<TextInputRN>();
  const passwordInputRef = useRef<TextInputRN>();

  const {
    loginRequest,
    isSignedIn,
    email: userEmail,
  } = useAppSelector((state) => ({
    loginRequest: state.auth.requests?.login,
    isSignedIn: state.auth.isSignedIn,
    email: state.auth.user?.email,
  }));

  useEffect(() => {
    if (isSignedIn && !!userEmail && isFocused) {
      identify();
      track('user_logged_in', { email: userEmail });

      if (route?.params?.onSuccess) {
        route.params.onSuccess(route.params);
        return;
      }
      navigation.goBack();
    }
  }, [isSignedIn, navigation, userEmail, route.params, isFocused, track, identify]);

  const [{ email, password }, setInput] = useState<LogIn>({
    email: '',
    password: '',
  });

  const [validationErrors, setValidationErrors] = useState<{ [field: string]: string[] }>({
    email: [],
    password: [],
  });

  const [
    containerAnimated,
    sectionMainAnimated,
    forgotPasswordButtonAnimated,
    loginButtonAnimated,
    createAccountButtonAnimated,
  ] = useLoginAnimation();

  useMount(() => {
    dispatch(authActions.clearRequests());
  });

  const handleLogIn = useCallback(() => {
    const { isValid, errors } = validate({ email, password }, loginRules);

    setValidationErrors(errors);

    if (isValid) {
      Keyboard.dismiss();

      dispatch(
        authActions.login.request({
          email,
          password,
        }),
      );
    }
  }, [dispatch, email, password]);

  const handleInput = (name: keyof LogIn, value: string) => {
    setInput((prevInput) => ({ ...prevInput, [name]: value }));
  };

  return (
    <SafeAreaView style={ApplicationStyles.loginContainer}>
      <Pressable
        style={styles.closeSection}
        onPress={() => {
          dispatch(authActions.setSignInState(true));
          navigation.goBack();
        }}>
        <CloseIcon color={Colors.white} accessible={true} accessibilityLabel="Close button" />
      </Pressable>
      <KeyboardAvoidingView style={styles.container} behavior={'padding'}>
        <Pressable onPress={Keyboard.dismiss} style={ApplicationStyles.flex} accessible={false}>
          <Animated.View style={[styles.keyboardAvoidingContent, containerAnimated]}>
            <Animated.View style={sectionMainAnimated}>
              <AppLogo style={styles.logo} accessible={true} accessibilityLabel="Logo" />
              <Text style={styles.textMain} accessible={true} accessibilityLabel="Text" testID="Text">
                Log into your Goal Zero app account or create a new app account. (This is the same account used for
                goalzero.com)
              </Text>
            </Animated.View>
            <TextInput
              accessible={true}
              accessibilityLabel="Email field"
              keyboardType="email-address"
              autoCapitalize="none"
              autoCorrect={false}
              placeholder="Email"
              textInputPlaceholder="me@email.com"
              value={email}
              isError={!!validationErrors.email.length}
              onChangeText={(value) => handleInput('email', value)}
              maxLength={AppConfig.maxTextInputLength}
              returnKeyType="next"
              ref={emailInputRef as TextInputRef}
              onSubmitEditing={() => {
                (passwordInputRef as TextInputRef)?.current?.focusNextInput();
              }}
            />
            <ValidationErrorList errorList={validationErrors.email} />
            <TextInputPassword
              containerStyle={styles.passwordInput}
              placeholder="Password"
              value={password}
              isError={!!validationErrors.password.length}
              onChangeText={(value) => handleInput('password', value)}
              maxLength={AppConfig.maxTextInputLength}
              returnKeyType={isIOS ? 'go' : 'done'}
              ref={passwordInputRef as TextInputRef}
              onSubmitEditing={handleLogIn}
            />
            <ValidationErrorList errorList={validationErrors.password} />
            <Animated.View style={[styles.forgotButton, forgotPasswordButtonAnimated]}>
              <ButtonSimple
                title="Forgot Password?"
                onPress={() => navigation.navigate('ForgotPassword')}
                accessibilityLabel="Forgot Password?"
              />
            </Animated.View>
            <Animated.View style={loginButtonAnimated}>
              <Button
                disabled={loginRequest}
                showLoading={loginRequest}
                title="Log In"
                onPress={handleLogIn}
                accessibilityLabel="Log In"
              />
            </Animated.View>
            <Animated.View style={createAccountButtonAnimated}>
              <Button
                inverse
                title="Create Account"
                onPress={() => navigation.navigate('CreateAccount')}
                accessibilityLabel="Create Account"
              />
            </Animated.View>
          </Animated.View>
        </Pressable>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginHorizontal: Metrics.baseMargin,
    marginBottom: Metrics.baseMargin,
  },
  keyboardAvoidingContent: {
    flex: 1,
    justifyContent: 'center',
  },
  closeSection: {
    width: 32,
    height: 32,
    marginLeft: Metrics.baseMargin,
    padding: Metrics.smallMargin,
  },
  logo: {
    alignSelf: 'center',
    marginBottom: Metrics.marginSection,
  },
  textSubMain: {
    ...Fonts.font.base.bodyOne,
    color: Colors.transparentWhite('0.87'),
  },
  textMain: {
    ...Fonts.font.base.bodyOne,
    color: Colors.transparentWhite('0.87'),
    textAlign: 'center',
  },
  passwordInput: {
    marginTop: Metrics.marginSection,
  },
  forgotButton: {
    alignSelf: 'flex-end',
  },
});

export default LogInScreen;
