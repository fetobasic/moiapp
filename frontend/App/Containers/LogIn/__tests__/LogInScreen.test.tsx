import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import LogInScreen from '../LogInScreen';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('LogInScreen', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
      },
      route: { params: {} },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <LogInScreen {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
