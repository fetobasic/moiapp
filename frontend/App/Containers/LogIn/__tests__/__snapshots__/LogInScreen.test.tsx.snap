// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInScreen renders screen 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 32,
        "marginLeft": 24,
        "padding": 8,
        "width": 32,
      }
    }
  >
    <SvgMock
      accessibilityLabel="Close button"
      accessible={true}
      color="rgba(255, 255, 255, 1)"
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
          "marginBottom": 24,
          "marginHorizontal": 24,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={false}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "justifyContent": "center",
            "transform": [
              {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "translateY": -32,
                },
              ],
            }
          }
        >
          <SvgMock
            accessibilityLabel="Logo"
            accessible={true}
            style={
              {
                "alignSelf": "center",
                "marginBottom": 16,
              }
            }
          />
          <Text
            accessibilityLabel="Text"
            accessible={true}
            style={
              {
                "color": "rgba(255, 255, 255, 0.87)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
                "textAlign": "center",
              }
            }
            testID="Text"
          >
            Log into your Goal Zero app account or create a new app account. (This is the same account used for goalzero.com)
          </Text>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(27, 27, 27, 1)",
              "elevation": 12,
              "height": 56,
              "shadowColor": "rgba(0, 0, 0, 1)",
              "shadowOffset": {
                "height": 4,
                "width": 4,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 12,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "borderColor": "rgba(54, 54, 54, 1)",
                "borderRadius": 8,
                "borderWidth": 1,
                "flex": 1,
                "height": 56,
                "justifyContent": "center",
              }
            }
          >
            <Text
              collapsable={false}
              numberOfLines={1}
              style={
                {
                  "backgroundColor": "rgba(27, 27, 27, 1)",
                  "color": "rgba(255, 255, 255, 0.871)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 14,
                  "fontWeight": "400",
                  "lineHeight": 20,
                  "marginLeft": 8,
                  "paddingHorizontal": 4,
                  "position": "absolute",
                  "top": 16,
                }
              }
            >
              Email
            </Text>
            <TextInput
              accessibilityLabel="Email field"
              accessible={true}
              autoCapitalize="none"
              autoCorrect={false}
              isError={false}
              keyboardAppearance="dark"
              keyboardType="email-address"
              maxFontSizeMultiplier={1.15}
              maxLength={50}
              numberOfLines={1}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onPressIn={[Function]}
              onSubmitEditing={[Function]}
              placeholder=""
              placeholderTextColor="rgba(255, 255, 255, 0.38)"
              returnKeyType="next"
              selectionColor="rgba(191, 210, 43, 1)"
              style={
                [
                  {
                    "color": "rgba(255, 255, 255, 0.87)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 16,
                    "paddingHorizontal": 12,
                  },
                  undefined,
                ]
              }
              textInputPlaceholder="me@email.com"
              underlineColorAndroid="rgba(0,0,0,0)"
              value=""
            />
          </View>
        </View>
        <View />
        <View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(27, 27, 27, 1)",
                "elevation": 12,
                "height": 56,
                "marginTop": 16,
                "shadowColor": "rgba(0, 0, 0, 1)",
                "shadowOffset": {
                  "height": 4,
                  "width": 4,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 12,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "borderColor": "rgba(54, 54, 54, 1)",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flex": 1,
                  "height": 56,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                collapsable={false}
                numberOfLines={1}
                style={
                  {
                    "backgroundColor": "rgba(27, 27, 27, 1)",
                    "color": "rgba(255, 255, 255, 0.871)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                    "marginLeft": 8,
                    "paddingHorizontal": 4,
                    "position": "absolute",
                    "top": 16,
                  }
                }
              >
                Password
              </Text>
              <TextInput
                accessibilityLabel="Password field"
                accessible={true}
                isError={false}
                keyboardAppearance="dark"
                maxFontSizeMultiplier={1.15}
                maxLength={50}
                numberOfLines={1}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onPressIn={[Function]}
                onSubmitEditing={[Function]}
                placeholder=""
                placeholderTextColor="rgba(255, 255, 255, 0.38)"
                returnKeyType="go"
                secureTextEntry={true}
                selectionColor="rgba(191, 210, 43, 1)"
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 16,
                      "paddingHorizontal": 12,
                    },
                    [
                      {
                        "paddingRight": 40,
                      },
                      undefined,
                    ],
                  ]
                }
                testID="passwordInput"
                underlineColorAndroid="rgba(0,0,0,0)"
                value=""
              />
            </View>
          </View>
        </View>
        <View />
        <View
          collapsable={false}
          style={
            {
              "alignSelf": "flex-end",
              "transform": [
                {
                  "translateY": 18,
                },
              ],
            }
          }
        >
          <View
            accessibilityLabel="Forgot Password?"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "opacity": 1,
                },
                [
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "rgba(191, 210, 43, 1)",
                    "fontFamily": "Roboto-Medium",
                    "fontSize": 14,
                    "fontWeight": "400",
                    "lineHeight": 20,
                  },
                ]
              }
            >
              Forgot Password?
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "translateY": 36,
                },
              ],
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "opacity": 1,
                },
                [
                  {
                    "elevation": 12,
                    "shadowColor": "rgba(0, 0, 0, 1)",
                    "shadowOffset": {
                      "height": 4,
                      "width": 4,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 12,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              accessibilityLabel="Log In"
              accessible={true}
              style={
                {
                  "borderRadius": 24,
                  "overflow": "hidden",
                }
              }
            >
              <RNSVGSvgView
                bbHeight="48"
                bbWidth="100%"
                focusable={false}
                height={48}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 48,
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGDefs>
                    <RNSVGLinearGradient
                      gradient={
                        [
                          0,
                          -13224394,
                          1,
                          1776411,
                        ]
                      }
                      gradientTransform={null}
                      gradientUnits={0}
                      name="grad"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0.5"
                    />
                  </RNSVGDefs>
                  <RNSVGRect
                    fill={
                      {
                        "brushRef": "grad",
                        "type": 1,
                      }
                    }
                    height="48"
                    propList={
                      [
                        "fill",
                      ]
                    }
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <View
                    style={
                      [
                        {
                          "borderRadius": 24,
                          "marginHorizontal": 1,
                          "marginTop": 1,
                        },
                        {
                          "backgroundColor": "rgba(191, 210, 43, 1)",
                          "height": 46,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "paddingVertical": 12,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "rgba(255, 255, 255, 1)",
                                "fontFamily": "Roboto-Medium",
                                "fontSize": 14,
                                "fontWeight": "700",
                                "lineHeight": 20,
                                "textTransform": "uppercase",
                              },
                              {
                                "color": "rgba(27, 27, 27, 1)",
                              },
                              undefined,
                            ]
                          }
                        >
                          Log In
                        </Text>
                      </View>
                    </View>
                  </View>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "translateY": 56,
                },
              ],
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "opacity": 1,
                },
                [
                  {
                    "elevation": 12,
                    "shadowColor": "rgba(0, 0, 0, 1)",
                    "shadowOffset": {
                      "height": 4,
                      "width": 4,
                    },
                    "shadowOpacity": 0.2,
                    "shadowRadius": 12,
                  },
                  undefined,
                ],
              ]
            }
          >
            <View
              accessibilityLabel="Create Account"
              accessible={true}
              style={
                {
                  "borderRadius": 24,
                  "overflow": "hidden",
                }
              }
            >
              <RNSVGSvgView
                bbHeight="48"
                bbWidth="100%"
                focusable={false}
                height={48}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 48,
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGDefs>
                    <RNSVGLinearGradient
                      gradient={
                        [
                          0,
                          -13224394,
                          1,
                          1776411,
                        ]
                      }
                      gradientTransform={null}
                      gradientUnits={0}
                      name="grad"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0.5"
                    />
                  </RNSVGDefs>
                  <RNSVGRect
                    fill={
                      {
                        "brushRef": "grad",
                        "type": 1,
                      }
                    }
                    height="48"
                    propList={
                      [
                        "fill",
                      ]
                    }
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <View
                    style={
                      [
                        {
                          "borderRadius": 24,
                          "marginHorizontal": 1,
                          "marginTop": 1,
                        },
                        {
                          "backgroundColor": "rgba(27, 27, 27, 1)",
                          "height": 46,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "paddingVertical": 12,
                          },
                          undefined,
                        ]
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "rgba(255, 255, 255, 1)",
                                "fontFamily": "Roboto-Medium",
                                "fontSize": 14,
                                "fontWeight": "700",
                                "lineHeight": 20,
                                "textTransform": "uppercase",
                              },
                              {
                                "color": "rgba(191, 210, 43, 1)",
                              },
                              undefined,
                            ]
                          }
                        >
                          Create Account
                        </Text>
                      </View>
                    </View>
                  </View>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
